Sources/Classes/core/Storage/MiniAppSecureStorage.swift

```diff
@@ -0,0 +1,256 @@
+                strongSelf.isBusy = false
+                completion?(.success(true))
+            }
+        }
+    }
+
+    // MARK: - Internal
+    private func loadLocalStorage() -> [String: String]? {
+        let secureStoragePath = MiniAppSecureStorage.miniAppPath(appId: appId)
+        let secureStorageName = MiniAppSecureStorage.storageName + ".plist"
+        let loadedStorage = FileManager.default.retrievePlist(secureStorageName, from: secureStoragePath, as: [String: String].self)
+        return loadedStorage
+    }
+
+    private static func miniAppPath(appId: String) -> URL {
+        return FileManager.getMiniAppDirectory(with: appId)
+    }
+
+    private static func storagePath(appId: String) -> URL {
+        return FileManager.getMiniAppDirectory(with: appId).appendingPathComponent("/\(storageName).plist")
+    }
+
+    private func saveStoreToDisk(completion: (() -> Void)? = nil) throws {
+        guard let storage = storage else { throw MiniAppSecureStorageError.storageNotExistent }
+        MiniAppLogger.d("🔑 Secure Storage: write store to disk")
+        let secureStoragePath = MiniAppSecureStorage.storagePath(appId: appId)
+        let secureStorageData = try PropertyListEncoder().encode(storage)
+        try secureStorageData.write(to: secureStoragePath, options: .completeFileProtectionUnlessOpen)
+        MiniAppLogger.d("🔑 Secure Storage: write store to disk completed")
+    }
+
+    // MARK: - Clear
+    static func clearSecureStorage() throws {
+        MiniAppLogger.d("🔑 Secure Storage: destroy")
+        let cachePath = FileManager.default.urls(for: .cachesDirectory, in: .userDomainMask)[0]
+        let miniAppPath = cachePath.appendingPathComponent("/MiniApp/")
+        guard let contentNames = try? FileManager.default.contentsOfDirectory(atPath: miniAppPath.path) else { return }
+        for name in contentNames {
+            let url = miniAppPath.appendingPathComponent("/" + name)
+            if let isDirectory = (try url.resourceValues(forKeys: [.isDirectoryKey])).isDirectory, isDirectory {
+                do {
+                    try FileManager.default.removeItem(at: url.appendingPathComponent("/" + storageFullName))
+                    MiniAppLogger.d("🔑 Secure Storage: destroyed storaged for \(name)")
+                } catch {
+                    MiniAppLogger.d("🔑 Secure Storage: could not destroy storaged for \(name)")
+                }
+            } else {
+                MiniAppLogger.d("🔑 Secure Storage: ignored \(name)")
+            }
+        }
+    }
+
+    public static func clearSecureStorage(for miniAppId: String) throws {
+        MiniAppLogger.d("🔑 Secure Storage: destroy")
+        try FileManager.default.removeItem(at: storagePath(appId: miniAppId))
+    }
+
+    public func size() throws -> UInt64 {
+        let fileSize = MiniAppSecureStorage.storagePath(appId: appId).fileSize
+        guard fileSize > 0 else { throw MiniAppSecureStorageError.storageFileEmpty }
```

Comment:
Would this be the case that the Secure Storage failed to load/hasn't finished loading yet?
Or is this the case where the user hasn't set any items yet?

I think the case that the user hasn't set any items, we don't need to return an error and we can just return 0.
And in the case that Secure Storage hasn't loaded, we can return some specific error like `SecureStorageNotLoaded`.

What do you think?

Reply:
@corycaywood the minimum size for the plist is around 42 bytes so it should always reflect if there is no store file yet (plist is in xml format so apple adds a few bits even if it's empty). So `0` is no file and `> 0` we've got a store. But sure I can remove the error check and just return the size no matter what the store doesn't need to be loaded for the size method. wdyt?